packages/tests/shadydom/event-path.html

```diff
@@ -938,20 +938,121 @@
       assert.isTrue(heardEvent);
     });
 
-    test('events dispatched to unassigned slottables bubble through the tree above them', () => {
-      const noSlot = document.createElement('x-no-slot');
-      const unassignedSlottable = document.createElement('div');
+    suite('unassigned slottable targets', () => {
+      test('events bubble through the tree above them', () => {
+        const noSlot = document.createElement('x-no-slot');
+        const unassignedSlottable = document.createElement('div');
 
-      ShadyDOM.wrapIfNeeded(document.body).appendChild(noSlot);
-      ShadyDOM.wrapIfNeeded(noSlot).appendChild(unassignedSlottable);
-      assert.equal(ShadyDOM.wrapIfNeeded(unassignedSlottable).assignedSlot, undefined);
+        ShadyDOM.wrap(document.body).appendChild(noSlot);
+        ShadyDOM.wrap(noSlot).appendChild(unassignedSlottable);
+        assert.strictEqual(ShadyDOM.wrap(unassignedSlottable).assignedSlot, null);
 
-      let listenerTriggered = false;
-      ShadyDOM.wrapIfNeeded(noSlot).addEventListener('some-event', () => listenerTriggered = true);
-      ShadyDOM.wrapIfNeeded(unassignedSlottable).dispatchEvent(new Event('some-event', {bubbles: true}));
-      assert.ok(listenerTriggered);
+        let listenerTriggered = false;
+        ShadyDOM.wrap(noSlot).addEventListener('some-event', () => listenerTriggered = true);
+        ShadyDOM.wrap(unassignedSlottable).dispatchEvent(new Event('some-event', {bubbles: true}));
+        assert.ok(listenerTriggered);
 
-      ShadyDOM.wrapIfNeeded(document.body).removeChild(noSlot);
+        ShadyDOM.wrap(document.body).removeChild(noSlot);
+      });
+
+      test('events have the correct phase during dispatch', () => {
+        /*
+          <x-no-slot> <!-- outerNoSlot -->
+            #shadow-root
+              <div> <!-- outerDiv -->
+                <x-no-slot> <!-- innerNoSlot -->
+                  #shadow-root
+                  #/shadow-root
+                  <div></div> <!-- target -->
+                </x-no-slot>
+              </div>
+            #/shadow-root
+          </x-no-slot>
+        */
+        const outerNoSlot = document.createElement('x-no-slot');
+        ShadyDOM.wrap(document.body).appendChild(outerNoSlot);
+        const outerDiv = document.createElement('div');
+        ShadyDOM.wrapIfNeeded(outerNoSlot).shadowRoot.appendChild(outerDiv);
+        const innerNoSlot = document.createElement('x-no-slot');
+        ShadyDOM.wrap(outerDiv).appendChild(innerNoSlot);
+        const target = document.createElement('div');
+        ShadyDOM.wrap(innerNoSlot).appendChild(target);
+
+        assert.strictEqual(ShadyDOM.wrap(target).assignedSlot, null);
+
+        const eventLog = [];
+        const pushToEventLog = (e) => {
+          eventLog.push({currentTarget: e.currentTarget, eventPhase: e.eventPhase});
+        };
+
+        // `ShadyDOM.wrapIfNeeded(outerNoSlot).shadowRoot` will also see this event in its path,
```

Comment:
Do we have enough information to correct this?

Reply:
I updated it so that it swaps descriptors for `eventPhase` (similar to existing code for `currentTarget`) when a listener on a shadow root is called. AFAICT, Shady DOM currently throws when dispatching to a shadow root itself, so I didn't handle that case. Also, the order is still wrong / depends on the order the listeners were added because listeners on shadow roots are actually added to their hosts, so they're all part of the same list. Fixing that seems like it's more effort than it's worth here / out of scope for this PR given most events go through native `dispatchEvent`.